src/glados/engine/tui_integration.py

```
@@ -0,0 +1,193 @@
+    This mixin extends the GLaDOS engine with methods to register TUI callbacks
+    and send real-time updates to the user interface for various engine events
+    such as TTS interruptions, chain of thought processes, and tool calling.
+    
+    Features:
+    - Thread-safe callback registration and execution
+    - Multiple event type support
+    - Error handling for TUI callback failures
+    - Enhanced method overrides with TUI notifications
+    """
+
+    def __init__(self, *args, **kwargs):
```

Ruff (0.11.9)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)